lib/exercises.rb

```diff
@@ -1,30 +1,115 @@
-
-
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(nmlogm) n for number of words, m for letters
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  letter_combos = Hash.new
+  strings.each do |word|
+    letters = word.chars.sort
+    sorted_letters = letters.join
+    if !letter_combos.has_key?(sorted_letters)
+      letter_combos[sorted_letters] = [word]
+    else letter_combos[sorted_letters] << word
+    end
+  end
+  return letter_combos.values
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(nlogn) [see two implementations below. Is sort_by faster than max_by?]
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
👍 If you look at the documentation `max_by` actually does sort the hash/list to figure out the max k elements.